olwebsite-app/src/components/Objects/Input/Input.jsx

```diff
@@ -1,14 +1,34 @@
 import React from "react";
 
-export default function Input({ className, label, type='text' }) {
+export default function Input({ 
+  className, 
+  label, 
+  onValueChange, 
+  errorMessage,
+  type='text'
+}) {
+ 
+  /************************************
+   * Helper Functions
+   ************************************/
+
+  function handleChange(event) {
+    onValueChange(event.target.value);
+  }
+
   /************************************
    * Render
    ************************************/
 
   return (
-    <label className={`${className}-input-container`}>
-        <p>{label}</p>
-        <input type={type}/>
-    </label>
+    <>
+      <label className={`${className}-input-container`}>
+          <div className='label-text-container'>
+            <p>{label}</p>
+            {errorMessage && <span>{errorMessage}</span>}
+          </div>
+          <input aria-label={`${label}-input`} type={type} onChange={handleChange}/>
```

Comment:
Why not use an inline function instead for onChange? Preference?

Reply:
Yeah I guess personal preference and readability, I could put it inline ya want